app/Traits/AccessControlAPI.php

```diff
@@ -60,15 +59,17 @@ function () use ($api_key) {
         );
     }
 
-    private function genericAccessControl(?string $api_key, ?string $fileset_hash, array $access_group_ids = [])
-    {
-
-        $cache_params = [$api_key, $fileset_hash, join('', $access_group_ids)];
+    private function genericAccessControl(
+        Collection $access_groups_by_api_key_user,
```

Comment:
rename to access_groups. Applies throughout the codebase. I added a few more comments below, but may have missed some

Reply:
@bradflood it has been renamed.